### PR TITLE
Split building test binary from other build checks.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: .github/workflows/pr-tests.yml [hunk 1]
```diff
@@ -55,19 +91,8 @@ jobs:
...
-    - name: Create tests archive
-      run: cargo nextest archive --archive-file tests.tar.zst --cargo-profile pr-tests --workspace --all-features
-    - name: Upload build artifacts
-      uses: actions/upload-artifact@v4
-      with:
-        name: tests_archive
-        path: |
-          tests.tar.zst
-          pil-stark-prover-deps.tar.zst
+      run: cargo check --all-targets --all --all-features --profile pr-tests
```
Comment (lvella) [inline .github/workflows/pr-tests.yml:94]: This looks redundant with:
```
cargo clippy --all --all-targets --all-features --profile pr-tests -- -D warnings
```

File: .github/workflows/pr-tests.yml [hunk 2]
```diff
@@ -22,6 +22,42 @@ jobs:
...
+      run: tar --zstd -cf pil-stark-prover-deps.tar.zst target/pr-tests/build/pil-stark-prover-*/out
+    - name: Upload build artifacts
+      uses: actions/upload-artifact@v4
+      with:
+        name: tests_archive
+        path: |
+          tests.tar.zst
+          pil-stark-prover-deps.tar.zst
+
+  build_check:
```
Comment (lvella) [inline .github/workflows/pr-tests.yml:58]: Maybe this should just be called "check".

**Review by lvella — CHANGES_REQUESTED** [review]
What I dislike about this is that there is a lot of redundant work done between the two jobs, as `cargo clippy` run is almost entirely included in `cargo build`.

Maybe the `--all-features` clippy run should go on the build job...

**Comment by lvella** [comment]
Looking at the execution times, even if there is redundant work, the split seems to pay off.

**Comment by chriseth** [comment]
I think the execution time does not include the waiting time, so maybe this does not pay off after all if we just have one build runner.

**Pull request closed without merging**